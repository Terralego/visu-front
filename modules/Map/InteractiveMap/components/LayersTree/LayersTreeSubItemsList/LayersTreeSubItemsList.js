import React from 'react';
import { RadioGroup, Radio } from '@blueprintjs/core';

export class LayersTreeSubItemsList extends React.Component {
  onSelectionChange = ({ target: { value: sublayer } }) => {
    const { layer, selectSublayer } = this.props;

    selectSublayer({ layer, sublayer: +sublayer });
  }

  render () {
    const { sublayers, layerState } = this.props;
    const { onSelectionChange } = this;
    const selectedValue = layerState.sublayers.reduce((value, selected, k) =>
      ((value === undefined || selected)
        ? +k
        : value), undefined);

    return (
      <div className="layers-tree-subitems-list">
        <RadioGroup
          onChange={onSelectionChange}
          selectedValue={selectedValue}
        >
          {sublayers.map((sublayer, k) => (
            <Radio
              key={sublayer.label}
              label={sublayer.label}
              value={k}
            />
          ))}
        </RadioGroup>
      </div>
    );
  }
}

export default LayersTreeSubItemsList;
