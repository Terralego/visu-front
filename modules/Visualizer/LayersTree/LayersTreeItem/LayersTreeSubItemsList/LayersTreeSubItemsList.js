import React from 'react';

import Radios from '../../../../Forms/Controls/Radios';
import Select from '../../../../Forms/Controls/Select';


export class LayersTreeSubItemsList extends React.Component {
  onSelectionChange = ({ target: { value: sublayer } }) => {
    const { layer, selectSublayer } = this.props;

    selectSublayer({ layer, sublayer: +sublayer });
  }

  onSelectionChangeWithSelect = sublayerindex => {
    const { layer, selectSublayer } = this.props;
    selectSublayer({
      layer,
      sublayer: sublayerindex,
    });
  }


  render () {
    const {
      sublayers,
      layerState,
    } = this.props;

    const { onSelectionChange } = this;

    const selectedValue = layerState.sublayers.reduce((sublayersValue, selected, k) =>
      ((sublayersValue === undefined || selected)
        ? +k
        : sublayersValue), undefined);

    return (
      <div className="layers-tree-subitems-list">
        {sublayers.length <= 5
          ? (
            <Radios
              onSelectionChange={onSelectionChange}
              selectedValue={selectedValue}
              sublayers={sublayers}
            />
          ) : (
            <Select
              label=""
              onChange={this.onSelectionChangeWithSelect}
              values={sublayers.map(({ label }, value) => ({
                label,
                value,
              }))}
              placeholder="Selectionner une variable"
              value={sublayers[selectedValue].label || 'Selectionner une variable'}
            />
          )
        }
      </div>
    );
  }
}

export default LayersTreeSubItemsList;
