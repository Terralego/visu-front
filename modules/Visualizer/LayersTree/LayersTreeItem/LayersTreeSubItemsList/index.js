import LayersTreeSubItemsList from './LayersTreeSubItemsList';
import { connectLayersTree } from '../../LayersTreeProvider/context';

export default connectLayersTree(({ selectSublayer, getLayerState }, { layer }) => ({
  layerState: getLayerState({ layer }),
  selectSublayer,
}))(LayersTreeSubItemsList);
