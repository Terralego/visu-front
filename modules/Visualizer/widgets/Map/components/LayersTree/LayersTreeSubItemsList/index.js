import LayersTreeSubItemsList from './LayersTreeSubItemsList';
import connectWidgetMap from '../../../connect';

export * from './LayersTreeSubItemsList';
export default connectWidgetMap(({ selectSublayer, getLayerState }, { layer }) => ({
  layerState: getLayerState({ layer }),
  selectSublayer,
}))(LayersTreeSubItemsList);
