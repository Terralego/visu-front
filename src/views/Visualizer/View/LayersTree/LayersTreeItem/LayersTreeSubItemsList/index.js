import LayersTreeSubItemsList from './LayersTreeSubItemsList';
import { connectView } from '../../../context';

export default connectView(({ selectSublayer, getLayerState }, { layer }) => ({
  layerState: getLayerState({ layer }),
  selectSublayer,
}))(LayersTreeSubItemsList);
