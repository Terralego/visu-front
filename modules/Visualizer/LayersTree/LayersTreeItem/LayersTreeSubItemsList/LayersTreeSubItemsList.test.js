import React from 'react';
import renderer from 'react-test-renderer';

import LayersTreeSubItemsList from './LayersTreeSubItemsList';


it('should render correctly sublayers list in radioboxes ', () => {
  const layer = {
    label: 'layer',
    sublayers: [{
      label: 'sublayer1', value: 'sublayer1',
    }, {
      label: 'sublayer2',
    }],
  };
  const tree = renderer.create((
    <div>
      <LayersTreeSubItemsList
        layer={layer}
        sublayers={layer.sublayers}
        layerState={{
          sublayers: [true, false],
        }}
      />
    </div>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly sublayers list in select ', () => {
  const layer = {
    label: 'layer',
    sublayers: [{
      label: 'sublayer1', value: 'sublayer1',
    }, {
      label: 'sublayer2', value: 'sublayer2',
    }, {
      label: 'sublayer3', value: 'sublayer3',
    }, {
      label: 'sublayer4', value: 'sublayer4',
    }, {
      label: 'sublayer5', value: 'sublayer5',
    }, {
      label: 'sublayer6', value: 'sublayer6',
    }],
  };
  const tree = renderer.create((
    <div>
      <LayersTreeSubItemsList
        layer={layer}
        sublayers={layer.sublayers}
        layerState={{
          sublayers: [true, false],
        }}
      />
    </div>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should change selection', () => {
  const layer = {
    label: 'layer',
    sublayers: [{
      label: 'sublayer1',
    }, {
      label: 'sublayer2',
    }],
  };
  const selectSublayer = jest.fn();
  const sublayerindex = 0;
  const instance = new LayersTreeSubItemsList({
    layer,
    sublayers: layer.sublayers,
    layerState: { sublayers: [true, false] },
    selectSublayer,
  });
  instance.onSelectionChange(sublayerindex);
  expect(selectSublayer).toHaveBeenCalledWith({
    layer, sublayer: 0,
  });
});
