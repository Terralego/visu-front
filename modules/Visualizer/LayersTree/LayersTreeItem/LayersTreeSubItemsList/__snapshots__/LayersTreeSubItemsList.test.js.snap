// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly sublayers list in radioboxes  1`] = `
<div>
  <div
    className="layers-tree-subitems-list"
  >
    <div>
      <label
        className="bp3-control bp3-radio"
      >
        <input
          checked={true}
          name="Blueprint3.RadioGroup-0"
          onChange={[Function]}
          type="radio"
          value={0}
        />
        <span
          className="bp3-control-indicator"
        />
        sublayer1
      </label>
      <label
        className="bp3-control bp3-radio"
      >
        <input
          checked={false}
          name="Blueprint3.RadioGroup-0"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        <span
          className="bp3-control-indicator"
        />
        sublayer2
      </label>
    </div>
  </div>
</div>
`;

exports[`should render correctly sublayers list in select  1`] = `
<div>
  <div
    className="layers-tree-subitems-list"
  >
    <div
      className="control-container"
      onKeyPress={[Function]}
      role="presentation"
    >
      <p
        className="control-label"
      >
        
      </p>
      <span
        className="bp3-popover-wrapper tf-select"
      >
        <span
          className="bp3-popover-target"
          onClick={[Function]}
        >
          <div
            className=""
            onKeyDown={[Function]}
          >
            <button
              className="bp3-button"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                className="bp3-button-text"
              >
                sublayer1
              </span>
              <span
                className="bp3-icon bp3-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    double-caret-vertical
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 0 0 .71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 0 0 5 7zm6 2H5a1.003 1.003 0 0 0-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0 0 11 9z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
        </span>
      </span>
    </div>
  </div>
</div>
`;
